onnx/defs/sequence/defs.cc

```diff
@@ -0,0 +1,235 @@
+// Copyright (c) ONNX Project Contributors.
+// Licensed under the MIT license.
+
+#include "onnx/defs/function.h"
+#include "onnx/defs/schema.h"
+
+namespace ONNX_NAMESPACE {
+static const char* SequenceEmpty_ver11_doc = R"DOC(
+
+)DOC";
+
+ONNX_OPERATOR_SET_SCHEMA(
+    SequenceEmpty,
+    11,
+    OpSchema()
+        .SetDoc(SequenceEmpty_ver11_doc)
+        .Attr(
+            "dtype",
+            "(Optional) The data type of the tensors in the output sequence. "
+            "The default type is 'float'.",
+            AttributeProto::INT,
+            OPTIONAL)
+        .Output(
+            0,
+            "output",
+            "Empty sequence.",
+            "S")
+        .TypeConstraint(
+            "S",
+            OpSchema::all_tensor_sequence_types(),
+            "Constrain output types to any tensor type.")
+        .TypeAndShapeInferenceFunction([](InferenceContext& ctx) {
+          // TODO: assign datatype here.
+        }));
+
+static const char* SequenceConstruct_ver11_doc = R"DOC(
+
+)DOC";
+
+ONNX_OPERATOR_SET_SCHEMA(
+    SequenceConstruct,
+    11,
+    OpSchema()
+        .SetDoc(SequenceConstruct_ver11_doc)
+        .Input(
+            0,
+            "inputs",
+            "List of tensors.",
+            "T",
+            OpSchema::Variadic)
+        .Output(
+            0,
+            "output_sequence",
+            "Sequence enclosing the input tensors.",
+            "S")
+        .TypeConstraint(
+            "T",
+            OpSchema::all_tensor_types(),
+            "Constrain input types to any tensor type.")
+        .TypeConstraint(
+            "S",
+            OpSchema::all_tensor_sequence_types(),
+            "Constrain output types to any tensor type.")
+        .TypeAndShapeInferenceFunction([](InferenceContext& ctx) {
+          propagateElemTypeFromInputToOutput(ctx, 0, 0);
+
+          auto numInputs = ctx.getNumInputs();
+          if (numInputs < 1 ||
+              !hasNInputShapes(ctx, static_cast<int>(numInputs))) {
+            return;
+          }
+
+          // Should we set shape info if input shapes are all known and are the same?
+        }));
+
+static const char* SequenceInsert_ver11_doc = R"DOC(
+
+)DOC";
+
+ONNX_OPERATOR_SET_SCHEMA(
+    SequenceInsert,
+    11,
+    OpSchema()
+        .SetDoc(SequenceInsert_ver11_doc)
+        .Input(
+            0,
+            "input_sequence",
+            "Input sequence.",
+            "S")
+        .Input(
+            1,
+            "tensor",
+            "Input tensor to be inserted into the input sequence.",
+            "T")
+        .Input(
+            2,
+            "position",
+            "Position in the sequence where the new tensor is inserted. "
+            "It is optional and default is to insert to the back of the sequence. "
+            "It must be a scalar(tensor of empty shape).",
```

Comment:
Are indices zero-based? Is the position required to be <= current length?

Reply:
Updated definition for position.